src/spyglass/lfp/v1/lfp.py

```diff
@@ -0,0 +1,911 @@
+    # Parameters for detecting LFP artifact times within a LFP group.
+    artifact_params_name: varchar(200)
+    ---
+    artifact_params: blob  # dictionary of parameters
+    """
+
+    def insert_default(self):
+        """Insert the default artifact parameters with an appropriate parameter dict."""
+        artifact_params = {}
+        artifact_params["zscore_thresh"] = None  # must be None or >= 0
+        artifact_params["amplitude_thresh"] = 500  # must be None or >= 0, ad units
+        # all electrodes of sort group
+        artifact_params["proportion_above_thresh"] = 0.1
+        artifact_params["removal_window_ms"] = 3.0  # in milliseconds
+        self.insert1(["default", artifact_params], skip_duplicates=True)
+
+        artifact_params_none = {}
+        artifact_params_none["zscore_thresh"] = None
+        artifact_params_none["amplitude_thresh"] = None
+        self.insert1(["none", artifact_params_none], skip_duplicates=True)
+
+
+@schema
+class LFPArtifactDetectionSelection(dj.Manual):
+    definition = """
+    # Specifies artifact detection parameters to apply to a sort group's recording.
+    -> LFP
+    -> LFPArtifactDetectionParameters
+    ---
+    custom_artifact_detection=0 : tinyint
+    """
+
+
+@schema
+class LFPArtifactDetection(dj.Computed):
+    definition = """
+    # Stores artifact times and valid no-artifact times as intervals.
+    -> LFPArtifactDetectionSelection
+    ---
+    artifact_times: longblob # np array of artifact intervals
+    artifact_removed_valid_times: longblob # np array of valid no-artifact intervals
+    artifact_removed_interval_list_name: varchar(200) # name of the array of no-artifact valid time intervals
+    """
+
+    def make(self, key):
+        if not (LFPArtifactDetectionSelection & key).fetch1(
+            "custom_artifact_detection"
+        ):
+            # get the dict of artifact params associated with this artifact_params_name
+            artifact_params = (LFPArtifactDetectionParameters & key).fetch1(
+                "artifact_params"
+            )
+
+            # recording_path = (SpikeSortingRecording & key).fetch1("recording_path")
+            # recording_name = SpikeSortingRecording._get_recording_name(key)
+            # recording = si.load_extractor(recording_path)
+
+            # job_kwargs = {"chunk_duration": "10s", "n_jobs": 4, "progress_bar": "True"}
+
+            # get LFP data
+            lfp_eseries = (LFP() & key).fetch_nwb()[0]["lfp"]
+
+            artifact_removed_valid_times, artifact_times = _get_artifact_times(
+                lfp_eseries,
+                key,
+                **artifact_params,
+            )
+
+            key["artifact_times"] = artifact_times
+            key["artifact_removed_valid_times"] = artifact_removed_valid_times
+
+            # set up a name for no-artifact times using recording id
+            # we need some name here for recording_name
+            key["artifact_removed_interval_list_name"] = (
```

Comment:
Due to the problem with long keys (see https://github.com/LorenFrankLab/spyglass/issues/453), we might need to cut down this name or use a hash (see https://docs.datajoint.org/python/definition/07-Primary-Key.html#using-hashes-as-primary-keys)